app/controllers/users_controller.rb

```diff
@@ -22,9 +22,47 @@ def create
     end
   end
 
+  def request_reset
+  end
+
+  def send_reset_link
+    user = User.where(email: params[:email]).first
+    if user
+      scheme = request.env['rack.url_scheme']
+      host = request.env['HTTP_HOST']
+      link = user.generate_reset_link(scheme, host)

```

Comment:
It should be something like `reset_link_url("token"). This from an app I am working on right now - http://cl.ly/image/3m361g2p3Y25